include/indexstore/indexstore.h

```diff
@@ -53,10 +53,14 @@
 #endif
 
 #ifndef INDEXSTORE_PUBLIC
-# if defined (_MSC_VER)
-#  define INDEXSTORE_PUBLIC __declspec(dllimport)
+# ifdef _WIN32
+#  ifdef IndexStore_EXPORTS
+#    define INDEXSTORE_PUBLIC __declspec(dllexport)
+#  else
+#    define INDEXSTORE_PUBLIC __declspec(dllimport)
+#  endif
 # else
-#  define INDEXSTORE_PUBLIC
+#  define INDEXSTORE_PUBLIC __attribute__((__visibility__("default")))
```

Comment:
After you removed the visibility options this has become redundant, since it's the default visibility.
I'd recommend to leave it blank as `#  define INDEXSTORE_PUBLIC` for consistency with the rest of llvm/clang.

Reply:
Sure.  Although, in ELF, it is possible to optimize this and use `__attribute__((__visibility__("default")))` when building and `__attribute__((__visibility__("protected")))` when using.